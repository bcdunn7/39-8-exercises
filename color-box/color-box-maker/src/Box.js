
function Box({
	id,
    color = 'blue', 
	width = 30, 
	height = 50,
	remove}) {
    const handleRemove = (e) => {
        remove(e.target.parentElement.id)
    }

  return (
	<div 
		id={id}
		style={{
			backgroundColor: `${color}`,
			width: `${width}px`,
			height: `${height}px`
		}}>
		<button onClick={(e) => handleRemove(e)}>X</button>
	</div>
  )
}

export default Box;