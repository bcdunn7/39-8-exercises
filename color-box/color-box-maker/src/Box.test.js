import React from "react";
import { render } from "@testing-library/react";
import Box from "./Box";

test("it renders without crashing", function() {
	render(<Box/>);
});

test("it matches snapshot", () => {
    const { asFragment } = render(<Box/>);
    expect(asFragment()).toMatchSnapshot();
})