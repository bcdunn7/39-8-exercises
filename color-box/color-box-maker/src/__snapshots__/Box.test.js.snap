// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="background-color: blue; width: 30px; height: 50px;"
  >
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
